src/server/hook_func.c

```diff
@@ -5535,6 +5581,12 @@ post_sendhookTPP(struct work_task *pwt)
 		}
 	}
 
+	pact->reply_expected &= ~(event);
+	if (failed_flag && check_for_latest_action(minfo, pact, j, event)) {
+		pact->action &= ~(event);
+		hook_track_save(minfo, j);
+	}
+
```

Comment:
What does the above do?

Reply:
So when the server receives a failure reply for a sent hook action, it checks if there is a same action marked for pending for the same hook while we were waiting for the reply. If yes then it removes the action pending bit from the current action object. as it will be taken care in the next sync operation due to the new pending mark in a fresh action pending object and we want to avoid duplicating the pending mark.
Hope I got through.

Reply:
Ok.